keyvault/client/src/main/java/com/azure/keyvault/SecretAsyncClientBuilder.java

```diff
@@ -0,0 +1,224 @@
+     * @return The updated Builder object.
+     * @throws NullPointerException if {@code logLevel} is {@code null}.
+     */
+    public SecretAsyncClientBuilder httpLogDetailLevel(HttpLogDetailLevel logLevel) {
+        Objects.requireNonNull(logLevel);
+        httpLogDetailLevel = logLevel;
+        return this;
+    }
+
+    /**
+     * Adds a policy to the set of existing policies that are executed after
+     * {@link SecretAsyncClient} required policies.
+     *
+     * @param policy The {@link HttpPipelinePolicy policy} to be added.
+     * @return The updated Builder object.
+     * @throws NullPointerException if {@code policy} is {@code null}.
+     */
+    public SecretAsyncClientBuilder addPolicy(HttpPipelinePolicy policy) {
+        Objects.requireNonNull(policy);
+        policies.add(policy);
+        return this;
+    }
+
+    /**
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
+     *
+     * @param client The HTTP client to use for requests.
+     * @return The updated Builder object.
+     * @throws NullPointerException If {@code client} is {@code null}.
+     */
+    public SecretAsyncClientBuilder httpClient(HttpClient client) {
+        Objects.requireNonNull(client);
+        this.httpClient = client;
+        return this;
+    }
+
+    /**
+     * Sets the HTTP pipeline to use for the service client.
+     *
+     * If {@code pipeline} is set, all other settings are ignored, aside from
+     * {@link SecretAsyncClientBuilder#vaultEndpoint(String) vaultEndpoint} to build {@link SecretAsyncClient}.
+     *
+     * @param pipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return The updated {@link SecretAsyncClientBuilder} object.
+     */
+    public SecretAsyncClientBuilder pipeline(HttpPipeline pipeline) {
+        Objects.requireNonNull(pipeline);
+        this.pipeline = pipeline;
+        return this;
+    }
+
+    private AsyncTokenCredentials getAsyncTokenCredentials() {
+        return new AsyncTokenCredentials("Bearer", credentials.authorizationHeaderValueAsync(new HttpRequest(HttpMethod.POST, vaultEndpoint)));
+    }
+
+    private class AsyncTokenCredentials implements AsyncServiceClientCredentials {
+
+        private String scheme;
+        private Mono<String> token;
+
+        AsyncTokenCredentials(String scheme, Mono<String> token) {
+            this.scheme = scheme;
+            this.token = token;
+        }
+
+        @Override
+        public Mono<String> authorizationHeaderValueAsync(HttpRequest httpRequest) {
+            if (scheme == null) {
+                scheme = "Bearer";
+            }
+            return token.flatMap(new Function<String, Mono<? extends String>>() {
+                @Override
+                public Mono<? extends String> apply(String tokenValue) {
+                    return Mono.just("Bearer " + tokenValue);
+                }
+            });
```

Comment:
This should be collapsed down into a lambda expression

Reply:
updated.